apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/schemas/Sgv2IndexesResourceImpl.java

```diff
@@ -59,7 +50,23 @@ public Uni<RestResponse<Object>> getAllIndexes(String keyspaceName, String table
             .where("table_name", Predicate.EQ, Values.of(tableName))
             .parameters(PARAMETERS_FOR_LOCAL_QUORUM)
             .build();
-    return fetchRowsAsync(query, true);
+
+    // check that we're authorized for the table
+    return authorizeSchemaReadsAsync(
+            Arrays.asList(
+                SchemaReads.table(keyspaceName, tableName, Schema.SchemaRead.SourceApi.REST)))
+        .onItem()
+        .invoke(
+            accessChecks -> {
+              if (!accessChecks.get(0).booleanValue()) {
+                throw new WebApplicationException(
+                    "Not authorized for read access to table '" + tableName + "'",
+                    Status.UNAUTHORIZED);
+              }
+            })
+        .onItem()
```

Comment:
Wait a bit.. Why are we having the `authorizeSchemaReadsAsync` method being used at all here? Imo this method should be removed, and everything should be going throught the schema manager? Would would not the `getTableAuthorized` not be used here? Imo remove the `authorizeSchemaReadsAsync` from the code and adapt where was used..

I have to say that this mix of `Async` methods existing in the resource base is very confusing.. Can you just not go directly to the schema manager? Or if you really need some extra processing, check how we created `CollectionManager` (table manager) and `NamespaceManager` (keyspace manager) that deal with tables and keyspaces..

Reply:
@ivansenic I am not 100% sure since this is code that come from previous implementation. I was wondering about same thing but assumed there was a reason (was first contemplating just removing it); possibly since actual read is from system tables, to get information about user tables' indexes. But same applies to many other places where no check is done so I guess I'll go and change handling here to use one of other methods.

As to confusion, I don't know. It is true that they have accumulated over time due to changes in handling but being very thin layer I am not sure I'd find them major distraction.
At this point I think I'd rather not spend more time on perfecting abstractions and get this iteration completed.

Reply:
@ivansenic Removed explicit call and using method that checks authorization as part of table access; then uses query against system table to get actual information.

Also added check for some basic failures (although not for unauthorized: wasn't sure how to trigger authorization failure easily).